### DISCUSSION TITLE
How to use images defined in webapp/src/images in webapp-libs

### DISCUSSION THREAD

**Discussion by Miheer22** [apptension/saas-boilerplate#617] [Q&A] [answered]
I am trying to use images defined in the `webapp/src/images/icons` directory within the `webapp-libs` module. But it turns out that we cannot import things from webapp to be used in webapp-core to be further used in any webapp-libs directory. 
Is there a way to use images defined in webapp into webapp-libs sub-directory?

**Comment by mkleszcz** [comment]
`webapp` package is the main package for the frontend app and is depends on (imports) other packages from `webapp-libs` directory. Packages from `webapp-libs` should not import anything from the root package, this is very easy way of the circular dependency issue:) If you would like to share code / assets between packages, use the `webapp-core` package and import things from this package or create a separate package in `webapp-libs`.

**Reply by Miheer22** [reply]
Hi @mkleszcz, thanks for replying. Actually there are a couple of images that I want to use in webapp-libs, but as you mentioned, even if I import something into webapp-core from webapp, that too is creating a circular dependency issue. So, should I create a copy of the images into webapp-core that I need to use in webapp-libs from webapp, along with the makeIcon.ts file in webapp, and then import these from webapp-core wherever I need to use them?

**Reply by mkleszcz** [reply, marked as answer]
Yes, exactly! I would try to move the icons to the `webapp-core` package and import them from it to the `webapp` and other `webapp-libs` packages.

**Reply by Miheer22** [reply]
Thanks a lot for the help and clarification @mkleszcz.
